### DISCUSSION TITLE
trying to create a command for a custom bounty

### DISCUSSION THREAD

**Discussion by SigmaLP** [ejektaflex/Bountiful#282] [Q&A] [answered]
First, I love this mod. We played around with it a bunch in 1.12 (I think) while trying to make a modpack and learn about configs.

I've got part of it figured out, where a command can run, but it still shows up blank/expired. 

/give @p bountiful:bounty{bounty_data:{},bounty_info:{}} 1

Even cleaning up what /kubejs hand had given me yielded an expired blank bounty.

`/give @p bountiful:bounty{bountiful:bountyboun_data:{objectives:[{id:all_obj_creeper,logicId:minecraft:entity,content:minecraft:creeper,amount:3,rarity:EPIC}],rewards:[{id:collector_rew_artifacts,logicId:minecraft:item,content:artifacts:universal_attractor,amount:1,rarity:UNCOMMON},{id:all_rew_entwine_rod,logicId:minecraft:item,content:architects_palette:entwine_rod,amount:21,rarity:RARE}]},bountiful:bounty_info:{rarity:RARE,timeStarted:45254,timeToComplete:233749}}
`

Any insight is greatly appreciated <3 

Sig

**Comment by ejektaflex** [comment, marked as answer]
Hi @SigmaLP !
`bounty_data` is not an NBT compound tag, it is an NBT string containing JSON data that is serialized with KotlinX Serialization. As such, it's not easy to do what you're asking - it's not made for easy editing. There aren't really any current plans to move away from this structure, since the average user has no reason to give themselves a bounty like this. The only use that I can think of is users who want to create their own specific bounties, which while nice is not the purpose of Bountiful :^) which is geared towards randomized bounty generation.

That might not be exactly what you want to hear, but I hope that clears it up!

**Reply by SigmaLP** [reply]
Ahhhhhhh! That makes sense. 


I was trying to incorporate some twitch integration using this mod. Thank you so much! I really appreciate your response and the enjoyment your mod has given me!

**Reply by ejektaflex** [reply]
For sure, thanks for asking!
